### PR TITLE
multiselect implementation

### LINKED ISSUE #245: Multi-select on space tokens' contextual menus
**Is your feature request related to a problem? Please describe.**
Whenever I'm managing space tokens on nodes, I find myself often working with several sides at once. The current method of applying multiple sides the same token requires individual assignment (right click, select Left, right click again, select Right).

**Describe the solution you'd like**
A multi-select method of choosing which properties you'd like to apply a token to. Since the menu already displays a checkbox to show if the token is currently applied, can that checkbox be interactive?

**Describe alternatives you've considered**
Another option might be a keyboard shortcut like Shift or Command that would allow for the selection of multiple properties at once, without closing the contextual menu.

**Additional context**
This probably only applies to spacing tokens since that's the one that has more than two options + an 'All', but maybe would have value in others too like Color. If a keyboard shortcut were to be used, it could be something that works for all tokens that have multiple properties without cluttering up the UI.

### PR DESCRIPTION
**Pull request by fokkerone**
### Add Multiselection for Contextmenu (Closes #245)
- 2 New TS Lint rules
- add DS_Store in .gitignore
- add Prettier as Formater, add some formating rules
- Solve LF/CRLF line ending problem for mdx stories
- Create 2 new exports in contextmenu and use these as alternative menu item in contextmenu

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (six7) [inline .prettierrc:1]: We already have a `.prettierrc.yml`, you can remove that

File: src/app/components/MoreButton/MoreButton.tsx [hunk 2]
```diff
@@ -47,32 +51,48 @@ export const MoreButton: React.FC<Props> = ({
...
-            key={property.name}
+          <MoreButtonPropertyMulti
+            key={`${property.name}-${property?.label}`}
             value={value}
             property={property}
             onClick={onClick}
           />
         ))}
+
+        {/* <ContextMenuSeparator /> */}
```
Comment (six7) [inline src/app/components/MoreButton/MoreButton.tsx:75]: This comment isn't needed, is it?

File: src/app/components/MoreButton/MoreButtonPropertyMulti.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const mainNodeSelectionValues = useSelector(mainNodeSelectionValuesSelector);
+  const isActive = React.useMemo(
+    () => mainNodeSelectionValues[property.name] === value,
+    [value, property, mainNodeSelectionValues],
+  );
+  const handleClick = React.useCallback(() => {
+    onClick(property, isActive);
+  }, [property, isActive, onClick]);
+
+  return (
```
Comment (six7) [inline src/app/components/MoreButton/MoreButtonPropertyMulti.tsx:34]: This current implementation doesn't work by using up/down arrow keys. It seems to just skip those multi entries.

File: .eslintrc.js [hunk 4]
```diff
@@ -1,17 +1,17 @@
 module.exports = {
-  extends: ['airbnb', 'airbnb-typescript', 'plugin:react-hooks/recommended'],
+  extends: ["airbnb", "airbnb-typescript", "plugin:react-hooks/recommended"],
```
Comment (six7) [inline .eslintrc.js:2]: can you revert those changes? we dont use double quotes in our prettier config.

**Review by six7 — CHANGES_REQUESTED** [review]
Overall nice! some minor things. Also would love to get @LiamMartens review on this as well from a typescript perspective.

File: .prettierrc (hunk 1 above)
Comment (fokkerone) [reply .prettierrc:1]: sorry forgot to delete after testing. will do!

File: src/app/components/MoreButton/MoreButton.tsx (hunk 2 above)
Comment (fokkerone) [reply src/app/components/MoreButton/MoreButton.tsx:75]: Nope

File: src/app/components/MoreButton/MoreButtonPropertyMulti.tsx (hunk 3 above)
Comment (fokkerone) [reply src/app/components/MoreButton/MoreButtonPropertyMulti.tsx:34]: ok, dont have this on my agenda. See what I can do to implement the default menu A11Y from radix on this one too

File: .eslintrc.js (hunk 4 above)
Comment (fokkerone) [reply .eslintrc.js:2]: you haven´t a uniform style https://github.com/six7/figma-tokens/blob/main/.eslintrc.js
releates to .prettierrc

Comment (LiamMartens) [reply .eslintrc.js:2]: I saw you added a `prettierrc` which is fine - but these should still change to single quotes I believe

File: src/stories/TokenButton.stories.mdx [hunk 5]
```diff
@@ -1,156 +1,215 @@
-import { Meta, Story, ArgsTable, Canvas, Preview } from '@storybook/addon-docs';
-import { Provider } from 'react-redux';
+import { Meta, Story, ArgsTable, Canvas, Preview } from "@storybook/addon-docs";
+import { Provider } from "react-redux";
 
-import { store } from '@/app/store';
-import { SingleToken } from '@/types/tokens';
-import { TokenTypes } from '@/constants/TokenTypes';
-import { BoxShadowTypes } from '@/constants/BoxShadowTypes';
+import { store } from "@/app/store";
```
Comment (LiamMartens) [inline src/stories/TokenButton.stories.mdx:4]: I guess these should also revert back to single quotes

**Review by LiamMartens — CHANGES_REQUESTED** [review]
Just formatting

**Comment by six7** [comment]
Closing this in favor of #969

**Pull request closed without merging**